lib/flag_shih_tzu.rb

```diff
@@ -242,118 +283,131 @@ def flag_keys(colmn = DEFAULT_COLUMN_NAME)
+          )
         end
-
-        true
+      else
+        # ActiveRecord gem may not have loaded yet?
+        warn(
+          %[FlagShihTzu#has_flags: Table "#{custom_table_name}" doesn't exist.  Have all migrations been run?]
+        ) if has_ar
+        return false
       end
 
-      def sql_condition_for_flag(flag, colmn, enabled = true, custom_table_name = self.table_name)
-        check_flag(flag, colmn)
+      true
+    end
 
-        if flag_options[colmn][:flag_query_mode] == :bit_operator
-          # use & bit operator directly in the SQL query.
-          # This has the drawback of not using an index on the flags colum.
-          "(#{custom_table_name}.#{colmn} & #{flag_mapping[colmn][flag]} = #{enabled ? flag_mapping[colmn][flag] : 0})"
-        elsif flag_options[colmn][:flag_query_mode] == :in_list
-          # use IN() operator in the SQL query.
-          # This has the drawback of becoming a big query when you have lots of flags.
-          neg = enabled ? "" : "not "
-          "(#{custom_table_name}.#{colmn} #{neg}in (#{sql_in_for_flag(flag, colmn).join(',')}))"
-        else
-          raise NoSuchFlagQueryModeException
-        end
+    def sql_condition_for_flag(flag, colmn, enabled = true, custom_table_name = table_name)
+      check_flag(flag, colmn)
+
+      if flag_options[colmn][:flag_query_mode] == :bit_operator
+        # use & bit operator directly in the SQL query.
+        # This has the drawback of not using an index on the flags colum.
+        %[(#{custom_table_name}.#{colmn} & #{flag_mapping[colmn][flag]} = #{enabled ? flag_mapping[colmn][flag] : 0})]
+      elsif flag_options[colmn][:flag_query_mode] == :in_list
+        # use IN() operator in the SQL query.
+        # This has the drawback of becoming a big query
+        #   when you have lots of flags.
+        neg = enabled ? "" : "not "
+        %[(#{custom_table_name}.#{colmn} #{neg}in (#{sql_in_for_flag(flag, colmn).join(",")}))]
+      else
+        raise NoSuchFlagQueryModeException
       end
+    end
 
-      # returns an array of integers suitable for a SQL IN statement.
-      def sql_in_for_flag(flag, colmn)
-        val = flag_mapping[colmn][flag]
-        flag_value_range_for_column(colmn).select {|i| i & val == val}
-      end
+    # returns an array of integers suitable for a SQL IN statement.
+    def sql_in_for_flag(flag, colmn)
+      val = flag_mapping[colmn][flag]
+      flag_value_range_for_column(colmn).select { |i| i & val == val }
+    end
 
-      def sql_set_for_flag(flag, colmn, enabled = true, custom_table_name = self.table_name)
-        check_flag(flag, colmn)
-        "#{colmn} = #{colmn} #{enabled ? "| " : "& ~" }#{flag_mapping[colmn][flag]}"
-      end
+    def sql_set_for_flag(flag, colmn, enabled = true, custom_table_name = table_name)
+      check_flag(flag, colmn)
+      "#{colmn} = #{colmn} #{enabled ? "| " : "& ~" }#{flag_mapping[colmn][flag]}"
```

Comment:
Line is too long. [82/80]<br>Prefer single-quoted strings inside interpolations.